### PR TITLE
Dev virtual sensor ship timeseries

### PR DESCRIPTION
**Pull request by kok-foong**
Updated all components in the dispersion stack to use the new time series client with reduced tables

### CODE REVIEW

**Comment by kok-foong** [comment]
https://github.com/cambridge-cares/TheWorldAvatar/pull/1324 should be merged first

File: JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/QueryClient.java [hunk 1]
```diff
@@ -416,31 +417,22 @@ void setMeasureIri(List<Ship> ships) {
 
     void updateTimeSeriesData(List<Ship> ships) {
         try (Connection conn = remoteRDBStoreClient.getConnection()) {
-            ships.stream().forEach(ship -> {
+            ships.stream().filter(ship -> ship.hasTimeSeries() && !ship.getTimestampList().isEmpty()).forEach(ship -> {
```
Comment (ChungTingLao) [inline JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/QueryClient.java:420]: This change breaks the case where ship data is provided by static JSON file. Please change the logic to accommodate both static JSON and live data stream cases.

**Review by ChungTingLao — CHANGES_REQUESTED** [review]
Logic in query client of ship input agent is currently broken for static JSON case based on my testing. Please change the logic to accommodate both static JSON and live data stream cases.

File: JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/QueryClient.java (hunk 1 above)
Comment (kok-foong) [reply JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/QueryClient.java]: the relevant commit is reverted, should be fixed

**Review by ChungTingLao — APPROVED** [review]
I have tested with ship data from JSON files and CSV files and they both work as intended.

**Pull request merged**